### PR TITLE
fix: determine duplicate function precedence based on extension

### PR DESCRIPTION
**Pull request by khendrikse**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes https://github.com/netlify/pod-compute/issues/462

Right now, the cli doesn't behave the same as production when it comes to duplicate function precedence. This should fix it. For more detail check the issue1

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

**Review by danez — COMMENTED** [review]
Can we add a test?

**Comment by khendrikse** [comment]
I will!

File: src/lib/functions/registry.mjs [hunk 1]
```diff
@@ -181,6 +184,18 @@ export class FunctionsRegistry {
     return await listFunctions(...args)
   }
 
+  // Check if a function has a duplicate with a preceding extension
+  hasPrecedingDuplicate({ mainFile, name }) {
+    const registeredDupe = this.functions.get(name)
```
Comment (danez) [inline src/lib/functions/registry.mjs:189]: Can we handle `undefined` here? I know that there is the `has` before calling this, but just to be sure, can we early exit if `registeredDupe` is undefined?

I tried to look at this function as it's own unit and right now it needs kind of a precondition (that there is a duplicate) that someone would need to be aware of when calling it.

```suggestion
    const registeredDupe = this.functions.get(name)
    if (!registeredDupe) {
        return false;
    }
```

**Review by danez — CHANGES_REQUESTED** [review]

File: src/lib/functions/registry.mjs (hunk 1 above)
Comment (khendrikse) [reply src/lib/functions/registry.mjs:189]: Good idea! ✅

**Review by danez — DISMISSED** [review]

File: src/lib/functions/registry.mjs [hunk 2]
```diff
@@ -212,7 +212,8 @@ export class FunctionsRegistry {
     await Promise.all(deletedFunctions.map((func) => this.unregisterFunction(func.name)))
 
     await Promise.all(
-      functions.map(async ({ displayName, mainFile, name, runtime: runtimeName }) => {
+      // We reverse the array so we get the right functions precedence
+      functions.reverse().map(async ({ displayName, mainFile, name, runtime: runtimeName }) => {
```
[the remark this answers is not in the document]
Comment (khendrikse) [reply src/lib/functions/registry.mjs:216]: Sure! I added an extra comment now :) zip-it-and-ship-it returns an array with the last functions preceding the previous. But we need to reverse that in the cli

[the remark this answers is not in the document]
Comment (danez) [reply src/lib/functions/registry.mjs:218]: Nice, so much simpler :)

**Review by danez — APPROVED** [review]

**Comment by danez** [comment]
This broke some tests.

**Comment by khendrikse** [comment]
> This broke some tests I think.

I'm really confused as to how I was able to merge this PR :/

**Pull request merged**